hatch/src/generators/mod.rs

```diff
@@ -0,0 +1,10 @@
+use project::Project;
+use hatch_error::HatchResult;
+use std::path::PathBuf;
+
+pub mod tup;
+
+pub trait Generator {
+  // TODO: this interface should take references
+  fn generate_assets(&self, project_path: PathBuf, project: &Project) -> HatchResult<()>;
+}
```

Comment:
This is a brand new trait! We have a trait that is intended to be used in dynamic dispatch for generator types.